contracts/contracts/token/WOETH.sol

```diff
@@ -80,9 +80,11 @@ contract WOETH is ERC4626, Governable, Initializable {
         external
         onlyGovernor
     {
-        //@dev TODO: we could implement a feature where if anyone sends OETH direclty to
-        // the contract, that we can let the governor transfer the excess of the token.
-        require(asset_ != address(asset()), "Cannot collect OETH");
+        if (asset_ == address(asset())) {
+            uint256 surplus = OETH(asset()).balanceOf(address(this)) - totalAssets();
```

Comment:
On a quick look, I'm a bit lost here. OETH is a rebasing token. wOETH's appreciation comes from the OETH yields (which are distributed during rebases with increasing balances). Won't this forgo all unrealized yields as well?

Reply:
Looks like it has an internal log of oethCredits (updating during wrap and unwrap) but let me check

Reply:
With [WOETH ignore donations PR](https://github.com/OriginProtocol/origin-dollar/pull/2106/files#diff-8bc11d78a06b7abf6bfae9f732a591fbd71cac02a935b1bc3f57ee04a22d8f41R119) WOETH keeps the balance of the internal credits (of the OETH token) that change on mint, deposit, redeem, withdraw. 

I think this approach is mathematically correct since the `surplus` figures out the difference between WOETH internal credits amount and OETH internal credits (for WOETH contract) multiplied by the credits per token - in other words operating with OETH token amounts. 

Surplus could also be calculated using another approach by finding the difference of internal credits tracked by WOETH and OETH and then multiplying it by the current credits per token ( to arrive to the OETH token surplus amount). 

I think in either of the approaches the un-realized rebase yields aren't left on the table, since as long as we do the correct math on the underlying credits per token. The rebase will take care of correct yield distribution.

Reply:
Still a spot on comment @shahthepro 🙏  These are the sort of checks we need to make to be sure the math works correctly.

We could add a test for this @clement-ux, where: 
- WOETH has initial supply of 100 OETH
- 2 OETH are transferred to the WOETH contract
- OETH rebases (say increases balance of OETH on WOETH contract by x2)
- governor should be able to transfer out 4 OETH and no more

Reply:
Thanks for the detailed explanation. Will check both PRs today to get more clarity on this.

Yeah, totally agree that we should be having unit and fork tests with different scenarios involving donations and rebases.

Reply:
I've added the test that @sparrowDom suggested in this [commit](https://github.com/OriginProtocol/origin-dollar/pull/2119/commits/e192d5175c592645ddf40e0a070f0b0d072aa877) and it passes.

Reply:
> Surplus could also be calculated using another approach by finding the difference of internal credits tracked by WOETH and OETH and then multiplying it by the current credits per token ( to arrive to the OETH token surplus amount).

Do you mean dividing instead of multiplying maybe? 

An implementation could look something like this: 
``` solidity
(uint256 creditsBalanceHighres, uint256 creditsPerTokenHighres, ) = OETH(asset()).creditsBalanceOfHighres(address(this));
uint256 surplus = (creditsBalanceHighres - oethCreditsHighres).divPrecisely(creditsPerTokenHighres);
```

This could consume less gas but can lead to rounding issues IMO.